### PR TITLE
pulling max bytes of data through the webhdfs api

### PR DESCRIPTION
**Pull request by MaddyDev**
Issue: When preview is clicked on files larger than 30MB, it does not load the file and hangs. 
Fix: Only load 30MB data when they click on preview. Specifying the length param while making the api request would achieve this.

### CODE REVIEW

File: extensions/mssql/src/objectExplorerNodeProvider/fileSources.ts [hunk 1]
```diff
@@ -181,7 +181,7 @@ export class HdfsFileSource implements IFileSource {
 	public readFile(path: string, maxBytes?: number): Promise<Buffer> {
 		return new Promise((resolve, reject) => {
 			let error: HdfsError = undefined;
-			let remoteFileStream = this.client.createReadStream(path);
+			let remoteFileStream = this.client.createReadStream(path, maxBytes ? {'length': maxBytes} : {});
```
Comment (kevcunnane) [inline extensions/mssql/src/objectExplorerNodeProvider/fileSources.ts]: So, I used the meter code below this to handle going over the max bytes. Did I miss that we could just specify a "stop after N bytes" option in the request args instead? How did you learn about this?

File: extensions/mssql/src/objectExplorerNodeProvider/fileSources.ts [hunk 2]
```diff
@@ -187,18 +188,23 @@ export class HdfsFileSource implements IFileSource {
...
 
+			let data: any[] = [];
 			if (maxBytes) {
 				remoteFileStream = remoteFileStream.pipe(meter(maxBytes));
 				remoteFileStream.on('error', (err) => {
 					error = <HdfsError>err;
 					if (error.message.includes('Stream exceeded specified max')) {
-						error.message = localize('maxSizeReached', 'File exceeds max size of ${0}', bytes(maxBytes));
+						// We have data > maxbytes, show we're truncating
+						let previewNote: string = '########################### NOTICE: This file has been truncated at '+ bytes(maxBytes) +' for preview ########################### \r\n';
```
Comment (kevcunnane) [inline extensions/mssql/src/objectExplorerNodeProvider/fileSources.ts]: 2 things:
1. Please put a #### string above/below the truncation notice so it takes up more space per Vicky's suggestion
```
#########################################################################################################
####################### NOTICE: This file has been truncated at 30 MB for preview ##################################
#########################################################################################################
```
2. The truncation section should be localized. Therefore, you'll want to 
    - Get the localized string & find its length
    - Use this to output the right number of #####s to show on the lines above/below this

File: extensions/mssql/src/objectExplorerNodeProvider/fileSources.ts [hunk 3]
```diff
@@ -187,18 +188,23 @@ export class HdfsFileSource implements IFileSource {
...
+						data.splice(0, 0, Buffer.from(previewNote, 'utf-8'));
+						vscode.window.showWarningMessage(localize('maxSizeReached', "Preview only loads {0} data of the file and truncates the rest.", bytes(maxBytes)));
```
Comment (kevcunnane) [inline extensions/mssql/src/objectExplorerNodeProvider/fileSources.ts]: Please change the message. Might want to check with PM, but I would suggest
```suggestion
						vscode.window.showWarningMessage(localize('maxSizeReached', "The previewed file {0} has been truncated - only the first {1} byes are shown.",  remotePath, bytes(maxBytes)));
```

File: extensions/mssql/src/objectExplorerNodeProvider/fileSources.ts (hunk 2 above)
Comment (MaddyDev) [reply extensions/mssql/src/objectExplorerNodeProvider/fileSources.ts]: Sure thing!

File: extensions/mssql/src/objectExplorerNodeProvider/fileSources.ts (hunk 3 above)
Comment (MaddyDev) [reply extensions/mssql/src/objectExplorerNodeProvider/fileSources.ts]: Checked with Rony, he suggested to add the same message we're appending to the text file i.e. "The file has been truncated at 30 MB for preview." Made the changes.

**Pull request merged**